django_yubin/models.py

```diff
@@ -127,29 +136,68 @@ def get_message_parser(self):
     def get_email_message(self):
         """
         Returns EmailMultiAlternatives or EmailMessage depending on whether the email is multipart or not.
+
+        Note that this should reconstruct all the kwargs of EmailMessage/EmailMultiAlternatives
+        to not lose information. See Github issue #66 for more information.
+
+        XXX: perhaps an inspect.signature combined with **kwargs approach could validate
+        against unexpected django changes?
         """
         msg = self.get_message_parser()
         to = self.to() or mailparser_utils.get_addresses(msg.to)
         cc = self.cc() or mailparser_utils.get_addresses(msg.cc)
-        bcc = self.bcc() or mailparser_utils.get_addresses(msg.bcc)
+        bcc = self.bcc()
+
+        # Process headers, but ignore address headers - these are processed explicitly.
+        headers = {
+            header: value
+            for header, value in msg.headers.items()
+            if header.lower() not in PARSED_HEADERS_TO_IGNORE
+        }
 
         Email = EmailMultiAlternatives if msg.text_html else EmailMessage
         email = Email(
             subject=msg.subject,
             body='\n'.join(msg.text_plain),
             from_email=mailparser_utils.get_address(msg.from_),
-            to=to, cc=cc, bcc=bcc,
+            to=to,
+            bcc=bcc,
+            headers=headers,
+            cc=cc,
+            reply_to=mailparser_utils.get_addresses(msg.reply_to),
         )
 
+        # set the multipart subtype
+        content_type = msg.headers["Content-Type"].split(";", 1)[0]  # discard boundary
+        main_type, subtype = content_type.split("/", 1)
+        if main_type == "multipart":
+            email.mixed_subtype = subtype
+
+        # FIXME: what if a third alternative is added? Perhaps processing the parsed
```

Comment:
Right now MailParser only considers text and html, everything else is considered not_managed:

![image](https://github.com/APSL/django-yubin/assets/2037027/6977aef7-d2ea-4873-9654-d151c5598bec)

I don't know how to create en email with not_managed parts in a "normal" use case. If some day we find one we will see how to approach it.